pkg/config/remote/uptane/client.go

```diff
@@ -85,32 +89,85 @@ func (c *Client) Update(response *pbgo.LatestConfigsResponse) error {
 	return c.verify()
 }
 
+// ConfigRootVersion returns the version of the config root.json file
+func (s *State) ConfigRootVersion() uint64 {
+	meta, found := s.ConfigState[metaRoot]
+	if !found {
+		return 0
+	}
+	return meta.Version
+}
+
+// ConfigSnapshotVersion returns the version of the config snapshot.json file
+func (s *State) ConfigSnapshotVersion() uint64 {
+	meta, found := s.ConfigState[metaSnapshot]
+	if !found {
+		return 0
+	}
+	return meta.Version
+}
+
+// DirectorRootVersion returns the version of the director root.json file
+func (s *State) DirectorRootVersion() uint64 {
+	meta, found := s.DirectorState[metaRoot]
+	if !found {
+		return 0
+	}
+	return meta.Version
+}
+
+// DirectorTargetsVersion returns the version of the director targets.json file
+func (s *State) DirectorTargetsVersion() uint64 {
+	meta, found := s.DirectorState[metaTargets]
+	if !found {
+		return 0
+	}
+	return meta.Version
+}
+
 // State returns the state of the uptane client
 func (c *Client) State() (State, error) {
-	c.Lock()
-	defer c.Unlock()
-	configRootVersion, err := c.configLocalStore.GetMetaVersion(metaRoot)
+	targets, err := c.Targets()
```

Comment:
This isn't used until the very end of the function, so it can be moved closer to where it is used safely.